README.md

```diff
@@ -88,43 +134,101 @@ async function main(): Promise<void> {
+The library provides an easy to use promisfied API for creating connection, session, sender and 
+receiver links. One can provide event handlers as optional properties while creating entities or 
+can use the standard event emitter pattern to add listeners to the created entities.
+
+For implementing the Event Emitter pattern, we considered different approaches like:
+**extending rhea's objects** (Connection, Session, Sender, Receiver). However, this has an overhead 
+of ensuring that there are no naming conflicts in the method names. Selecting different names for 
+operations like `open` and `close` would add more confusion.
+
+Hence we settled with, **wrapping rhea's objects** under our own equivalent objects. To ensure that 
+user's can efficiently use different event emitter methods like `.once()`, `.on()`,
+`.prependListeners()`, etc. we decided to add listeners to rhea's objects 
+(connection, session, sender, receiver) for all the possible events that those objects can 
+emit. From those listeners, equivalent `rhea-promise` objects emit the same events with same 
+arguments as one would expect from an event emitted by one of the above mentioned objects from 
+`rhea`. User's can add/remove listeners to `rhea-promise` objects. Whenever an event is emitted 
+from an object from `rhea`, it's equivalent counterpart in `rhea-romise` would emit the same event.
+
+`rhea` deals little differently with *_error and *_close events. If an event listener is **not added** 
+to a link (sender, receiver) then it will emit the `sender_error, sender_close, receiver_error, receiver_close` 
+events to the `session` object that the `link` belongs to. If there are no event listeners 
+for thoe events at the `session` level then, it emits those events to the `connection` object. Still 
+if it could not find listeners for those events then the events are bubbled up to the `container` 
+object and eventually those events are transformed into `error` events, if there were no event 
+listeners for those events at the container level. 
+
+It does the same thing for `session_error, session_close` events. Events are bubbled up to the 
+`connection`  and `container` objects and eventually emitted as `error` events if no listeners 
+were found for those events at the respective levels.
+
+This behavior of `rhea` will not be reciprocated by `rhea-promise` since, it adds event listeners 
+to each of the equivalent rhea objects. Thus *_error and *_close events will never bubble up. 
+If you do not add event listeners for those objects while using `rhea-promise` then you will not 
+know why an `error` or a `close` event occurred. So please make sure that right kind of event 
+listeners are added to the right objects. This enforces good practices to be followed while using 
+the event emitter pattern.
+
+
+## Building the library
+- Clone the repo
+```
+git clone https://github.com/amqp/rhea-promise.git
+```
+- Install typescript, ts-node globally
+```
+npm i -g typescript
```

Comment:
why not have typescript and ts-node as dev dependencies that can be run through npm scripts?

Reply:
I do have them as dev dependencies. However it is incredibly useful to install them globally as well. As a dev, I have been using the ts-node repl a lot these days. Hence added those steps in the larger interest.